common/src/main/java/io/netty/util/AsciiString.java

```diff
@@ -188,34 +191,22 @@ public ByteString newInstance(byte[] value, int start, int length, boolean copy)
      * {@link CharSequence}s into the same {@link TextHeaders}.
      */
     public static int caseInsensitiveHashCode(CharSequence value) {
-        if (value instanceof AsciiString) {
-            try {
-                ByteProcessor processor = new ByteProcessor() {
-                    private int hash;
-                    @Override
-                    public boolean process(byte value) throws Exception {
-                        hash = hash * HASH_CODE_PRIME ^ toLowerCase(value) & HASH_CODE_PRIME;
-                        return true;
-                    }
-
-                    @Override
-                    public int hashCode() {
-                        return hash;
-                    }
-                };
-                ((AsciiString) value).forEachByte(processor);
-                return processor.hashCode();
-            } catch (Exception e) {
-                PlatformDependent.throwException(e);
+        if (value == null) {
+            return CASE_INSENSITIVE_HASHER.emptyHashValue();
+        }
+        if (value.getClass() == AsciiString.class) {
```

Comment:
`AsciiString` is a final class and equals should determine what we need in equality check. `instanceof` *may be* a O(n) type operation (not sure if JIT or compiler is smart enough to optimize in this case) that inspects the class hierarchy.